flake.nix

```diff
@@ -26,17 +29,53 @@
         let
           pkgs = nixpkgs.legacyPackages.${system};
           warnToUpgradeNix = pkgs.lib.warn "Please upgrade Nix to 2.7 or later.";
+          version = pkgs.lib.removeSuffix "\n" (builtins.readFile ./VERSION);
+          channels =
+            let
+              allChannels = (import "${nixos-org-configurations}/channels.nix").channels;
+              filterChannels = 
+                pkgs.lib.filterAttrs
+                  (n: v:
+                    builtins.elem v.status ["beta" "stable" "rolling"] &&
+                    pkgs.lib.hasPrefix "nixos-" n &&
+                    v ? variant && v.variant == "primary"
+                  )
+                  allChannels;
+            in
+              builtins.map
+                (n: let v = filterChannels.${n}; in
+                  {
+                    id = pkgs.lib.removePrefix "nixos-" n;
+                    title = pkgs.lib.removePrefix "nixos-" n;
+                    status = v.status;
+                    jobset =
+                      builtins.concatStringsSep
+                        "/"
+                        (pkgs.lib.init (pkgs.lib.splitString "/" v.job));
```

Comment:
Do you have plans to use the `jobset` in the future? Also, would `title` ever be different from `id`?

Reply:
> Do you have plans to use the `jobset` in the future?

Yes, showing what is in the binary cache. This used to work, but we was a little unstable and we removed it until we figured out other stuff first.

> Also, would `title` ever be different from `id`?

good question. maybe title is not needed here, but we would calculate in the frontend. I'll remove it for now.